packages/grid/x-data-grid/src/components/columnsManagement/GridColumnsManagement.tsx

```diff
@@ -0,0 +1,356 @@
+    const togglableColumns = getTogglableColumns ? getTogglableColumns(sortedColumns) : null;
+
+    const togglableSortedColumns = togglableColumns
+      ? sortedColumns.filter(({ field }) => togglableColumns.includes(field))
+      : sortedColumns;
+
+    if (!searchValue) {
+      return togglableSortedColumns;
+    }
+
+    return togglableSortedColumns.filter((column) =>
+      searchPredicate(column, searchValue.toLowerCase()),
+    );
+  }, [sortedColumns, searchValue, searchPredicate, getTogglableColumns]);
+
+  const hideableColumns = React.useMemo(
+    () => currentColumns.filter((col) => col.hideable),
+    [currentColumns],
+  );
+
+  const allHideableColumnsVisible = React.useMemo(
+    () =>
+      hideableColumns.every(
+        (column) =>
+          columnVisibilityModel[column.field] == null ||
+          columnVisibilityModel[column.field] !== false,
+      ),
+    [columnVisibilityModel, hideableColumns],
+  );
+
+  const allHideableColumnsHidden = React.useMemo(
+    () => hideableColumns.every((column) => columnVisibilityModel[column.field] === false),
+    [columnVisibilityModel, hideableColumns],
+  );
+
+  const firstSwitchRef = React.useRef<HTMLInputElement>(null);
+
+  React.useEffect(() => {
+    if (autoFocusSearchField) {
+      searchInputRef.current!.focus();
+    } else if (firstSwitchRef.current && typeof firstSwitchRef.current.focus === 'function') {
+      firstSwitchRef.current.focus();
+    }
+  }, [autoFocusSearchField]);
+
+  let firstHideableColumnFound = false;
+  const isFirstHideableColumn = (column: GridColDef) => {
+    if (firstHideableColumnFound === false && column.hideable !== false) {
+      firstHideableColumnFound = true;
+      return true;
+    }
+    return false;
+  };
+
+  return (
+    <React.Fragment>
+      <GridColumnsManagementHeader className={classes.header} ownerState={rootProps}>
+        <rootProps.slots.baseTextField
+          placeholder={apiRef.current.getLocaleText('columnsManagementSearchTitle')}
+          inputRef={searchInputRef}
+          value={searchValue}
+          onChange={handleSearchValueChange}
+          variant="outlined"
+          size="small"
+          InputProps={{
+            startAdornment: <rootProps.slots.quickFilterIcon sx={{ pr: 1 }} />,
+          }}
+          fullWidth
+          {...rootProps.slotProps?.baseTextField}
+        />
+      </GridColumnsManagementHeader>
+      <GridColumnsManagementBody className={classes.root} ownerState={rootProps}>
+        {currentColumns.map((column) => (
+          <FormControlLabel
+            key={column.field}
+            className={classes.row}
+            control={
+              <rootProps.slots.baseCheckbox
+                disabled={column.hideable === false}
+                checked={columnVisibilityModel[column.field] !== false}
+                onClick={toggleColumn}
+                name={column.field}
+                size="small"
+                sx={{ py: 0.5 }}
```

Comment:
Should we keep the normal size here?

```suggestion
                sx={{ p: 0.5 }}
```

<img width="326" src="https://github.com/mui/mui-x/assets/13808724/a602ea69-c3ef-426b-9614-f67c6962b983">

Reply:
I'm fine with using normal variant. Though, to me personally, normal size looks a bit heavier, in [designs](https://www.figma.com/file/tOAZwAT1ebcTH4tb4zv3Lo/Pivoting?type=design&node-id=991-16469&mode=design&t=6bBq2vLyGZvJuYPB-0) we have a small variant too.
<img width="160" alt="image" src="https://github.com/mui/mui-x/assets/12609561/bd692b46-5c96-473c-abe3-25688ff0e868">

Let me know what you think.